common/transaction_holder.cpp

```diff
@@ -41,7 +45,8 @@ TransactionHolder::~TransactionHolder() {
 }
 
 void TransactionHolder::SetTransaction(const ConfigurationPtr config, Transaction* txn) {
-  ExtractKeysInPartition(keys_in_partition_, config, *txn);
+  // TODO: involved_partitions_ is only needed by MH and SH, could avoid computing for LO
+  ExtractKeyPartitions(keys_in_partition_, involved_partitions_, config, *txn);
```

Comment:
not required for now but we should also empty `keys_in_partition_` and `involved_partitions_` here before adding new elements